### PR TITLE
ADR for mutable shared states in the frontend

### PR DESCRIPTION
**Pull request by HaGuesto**

### CODE REVIEW

**Comment by HaGuesto** [comment]
[2.0 Research state management options (Apollo Client, React Context and useContext) for base context - should we use apollo local or continue with react context (current) to save states? #sprint-jan-30 #sprint-feb-09](https://trello.com/c/JPkYs0QW/1102-20-research-state-management-options-apollo-client-react-context-and-usecontext-for-base-context-should-we-use-apollo-local-or-c)

File: docs/adr/adr_frontend_global_state.md
Comment (pylipp) [inline docs/adr/adr_frontend_global_state.md:34]: @HaGuesto what is the decision? Keeping everything the same because we don't want to refactor, and then try Apollo Reactive when a new situation with mutable shared space comes up?
Who else should review this doc?

Comment (HaGuesto) [reply docs/adr/adr_frontend_global_state.md]: I put more info into the decision and added Roanna and Vahid to review.

**Review by pylipp — APPROVED** [review]

**Comment by aerinsol** [comment]
I think the problem statement is fine, though it could be more concise. However, we have no idea of the pros/cons of each, nor of the cost for implementation or prototyping. I would not approve this for now.

**Comment by HaGuesto** [comment]
@aerinsol I added more information and Pros and Cons on it

**Comment by HaGuesto** [comment]
@aerinsol can we merge this after my updates? just cleaning up atm.

**Comment by aerinsol** [comment]
Let's add in the discussions from Slack as well to track, and then merge

**Comment by HaGuesto** [comment]
@aerinsol I do not find the discussions on slack anymore ...

**Comment by HaGuesto** [comment]
@aerinsol either we merge or close this PR. Their is no relevant to take here anymore. The ADR might be useful to have the related research taken down.

All the options mentioned in this PR are viable and we should not decide for one, but can use the appropriate one depending on the situation.

**Comment by aerinsol** [comment]
I will check this and either merge or close by EOW.

**Review by aerinsol — APPROVED** [review]
I approve. Did we end up trying out Apollo State?

**Comment by HaGuesto** [comment]
> I approve. Did we end up trying out Apollo State? 

Yes, quite nice actually. We are using it for the isOpen State of some oberlays

**Comment by aerinsol** [comment]
I do feel like we are using cacheing more and more and would like to
revisit the question of approach in terms of maintainability / performance
at a later date.

**Pull request merged**
